### DISCUSSION TITLE
[loki explore]: Hard to navigate around inside a selected time range, when many lines are returned

### DISCUSSION THREAD

**Discussion by iainlane** [grafana/grafana#55935] [Observability]
## Problem

This is probably the most common way for me to use Loki. I'll write a query, something quite general like `{namespace="foo", cluster="bar"}` and I'll narrow it down to a few-minute time window that I want to drill down into by selecting a range in the logs volume graph.

More often than not this is more data than Explore will load into one screen, so I'll get a message like

```
Line limit: 1000 reached, received logs cover 18.71% (31sec) of your selected time range (2min 45sec)
```

So now what I want to do is scroll around inside this selected range. Let's say the end of the interesting range is showing at the top of the screen now (older logs are shown first).

I press "Older logs"

![image](https://user-images.githubusercontent.com/321014/192737547-01364a1f-de08-4700-b35b-ce9335b8fa79.png)

to go back in time and see the log lines just before these ones. Now, two things happen that break my flow a bit:

  1. The time range I selected is stepped back, I can see logs from before its start.
  2. The whole set of logs is replaced, we show all of the next oldest 1000 log lines. All of the logs I was just looking at are gone now.

If the interesting logs are in the middle of one of the 'page breaks' this is frustrating as I can't see the whole context of my problem at once. Maybe I only wanted 10 more lines and 50 good ones were on the first page, there's no way to move the 'window' in that way.

The only way I know to work around this is to select a different time range that lands my target logs in the middle. That can be made more difficult by problem (1) in that the original time window I selected is now lost.

## Suggestion

I think some kind of 'pageless' (dynamic fetching of logs on scroll) would be really really great to address this problem. Then I could scroll the view up and down and focus very easily on what I need to see at that time.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
